tonic/src/codec/compression.rs

```diff
@@ -2,20 +2,30 @@ use crate::{metadata::MetadataValue, Status};
 use bytes::{Buf, BytesMut};
 #[cfg(feature = "gzip")]
 use flate2::read::{GzDecoder, GzEncoder};
+use http::HeaderValue;
 use std::fmt;
+use strum::EnumCount;
+use strum_macros::EnumCount as EnumCountMacro;
 #[cfg(feature = "zstd")]
 use zstd::stream::read::{Decoder, Encoder};
 
 pub(crate) const ENCODING_HEADER: &str = "grpc-encoding";
 pub(crate) const ACCEPT_ENCODING_HEADER: &str = "grpc-accept-encoding";
 
+/// This should always match the cardinality of `CompressionEncoding`
+pub(crate) const COMPRESSION_ENCODINGS_LENGTH: usize = CompressionEncoding::COUNT;
+
 /// Struct used to configure which encodings are enabled on a server or channel.
+/// Supports setting the priority of each compression
 #[derive(Debug, Default, Clone, Copy)]
 pub struct EnabledCompressionEncodings {
+    // We keep the encodings as struct values so we get `const fn` compatibility
```

Comment:
This all seems pretty roundabout. Can we just change `EnabledCompressionEncodings` to store a `Vec<CompressionEncoding>` instead?

Reply:
Is a slice (as is) ok, or would you prefer a true vec?

Reply:
The way you've implemented the array here seems to cause a bunch of unnecessary complexity. I think a `Vec` would be simpler to use internally (just `contains()` calls on it, `push`, `remove`).

I also think the first element in the list should take priority rather than the last, and not the most recently added element, that seems confusing to me.

Reply:
The way it's implemented, the most recently added element takes the **lowest** real priority; i.e the way it is currently implemented already matches your request of "the first element in the list should take priority rather than the last, and not the most recently added element". 

I will try to simplify a bit (with vec), but no matter the change, you will ultimately have a disconnect between where you insert things, and what the "priority" is

Reply:
The `array` instead of `Vec` was, originally, to localize/minimize the impact of my changes, and support a `copy` type; moving to Vec means spreading `.clone()` in a few places, but also moving to using refs in others

Reply:
See my other comments. Array seems okay to me.